### PR TITLE
[CFM-1137] -Group dashboard activity stream

### PR DESCRIPTION
**Pull request by ybaras**
https://issuetracker.amplexor.com/jira/browse/CFM-1137

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 1]
```diff
@@ -711,11 +715,28 @@ function c4m_message_format_operation(Message $message) {
...
 /**
+ * Return the photoalbum title of the inserted photo.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   The photoalbum if exists or an empty string.
+ */
+function c4m_message_get_photoalbum(Message $message) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: Rename the function to:
`c4m_message_format_photo_photoalbum`

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 2]
```diff
@@ -711,11 +715,28 @@ function c4m_message_format_operation(Message $message) {
...
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $node = $wrapper->field_node->value();
+  $photo_wrapper = entity_metadata_wrapper('node', $node);
+  $albumlabel = $photo_wrapper->node_gallery_ref_1->label();
+  $output = ($albumlabel) ? 'to Photo album ' . $albumlabel : '';
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: remove this line.

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 3]
```diff
@@ -711,11 +715,28 @@ function c4m_message_format_operation(Message $message) {
...
+ *   The Message entity.
+ *
+ * @return string
+ *   The photoalbum if exists or an empty string.
+ */
+function c4m_message_get_photoalbum(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $node = $wrapper->field_node->value();
+  $photo_wrapper = entity_metadata_wrapper('node', $node);
+  $albumlabel = $photo_wrapper->node_gallery_ref_1->label();
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: Try using:
```PHP
$albumlabel = $wrapper->field_node->node_gallery_ref_1->label();
```

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 4]
```diff
@@ -711,11 +715,28 @@ function c4m_message_format_operation(Message $message) {
...
+  $output = ($albumlabel) ? 'to Photo album ' . $albumlabel : '';
+  return $output;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]:
```PHP
return $album_label ? ' to photo album ' . $album_label : '';
```

**Comment by ordavidil** [comment]
@ybaras 
please add screenshots, thanks

**Comment by ybaras** [comment]
Photo in an album:

alb

Photo not in an album:

noalb

**Comment by ybaras** [comment]
alb

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 5]
```diff
@@ -711,11 +715,25 @@ function c4m_message_format_operation(Message $message) {
...
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   The photoalbum if exists or an empty string.
+ */
+function c4m_message_format_photo_photoalbum(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $album_label = $wrapper->field_node->node_gallery_ref_1->label();
+  return $album_label ? 'to Photo album ' . $album_label : '';
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: Add line spacing above the `return`

**Comment by ybaras** [comment]
Update all group activity stream texts according to the design and wireframes:

longshot

**Comment by ybaras** [comment]
**Before:**

before


**After:**

duptitle

**Pull request merged**